ARCHITECTURE.md

```diff
@@ -27,49 +27,66 @@ AI Town is split into a few layers:
-- Conversation memberships (`convex/game/conversationMembers.ts`) indicate that a player is a member
-  of a conversation. Players may only be in one active conversation at any point in time, and conversations
-  currently have exactly two members. Memberships may be in one of four states:
+- Worlds (`convex/aiTown/world.ts`) represent a map with many players interacting together.
+- Players (`convex/aiTown/player.ts`) are the core characters in the game. Players have human readable names and
+  descriptions, and they may be associated with a human user. At any point in time, a player may be pathfinding
+  towards some destination and has a current location.
+- Conversations (`convex/aiTown/conversations.ts`) are created by a player and end at some point in time.
+- Conversation memberships (`convex/aiTown/conversationMembership.ts`) indicate that a player is a member
+  of a conversation. Players may only be in one conversation at any point in time, and conversations
+  currently have exactly two members. Memberships may be in one of three states:
   - `invited`: The player has been invited to the conversation but hasn't accepted yet.
   - `walkingOver`: The player has accepted the invite to the conversation but is too far away to talk. The
     player will automatically join the conversation when they get close enough.
   - `participating`: The player is actively participating in the conversation.
-  - `left`: The player has left the conversation, and we keep the row around for historical queries.
 
-### Inputs (`convex/game/inputs.ts`)
+### Schema
+
+To keep the game state small and efficient to read and write, we store this data model across a few tables.
+
+- `worlds`: This table has a single document that stores all players, conversations, and agents. This
+  data is small and changes regularly over time.
+- `worldStatus`: Worlds can be started or stopped by the developer or paused for inactivity, and this
+  infrequently changing document tracks this world state.
+- `maps`: This table contains the map data for a given world. Since it's a bit larger than the player
+  state and infrequently changes, we store it in a separate table.
+- `playerDescriptions` and `agentDescriptions`: Human readable text describing players and agents that's
+  stored in separate tables, just like `maps`.
+- `archivedPlayers`, `archivedConversations`, `archivedAgents`: The game engine doesn't want to track
+  players that have left or conversations that are over, since it wants to keep its managed state small.
+  However, we may want to look at old conversations in the UI or from the agent code. So, whenever we
+  delete an entry from within the world's document, we "archive" it within these tables.
+- `participatedTogether`: The agent layer wants to know what the last conversation was between two players,
+  so this table represent a labelled graph indicating which players have talked to each other.
```

Comment:
I appreciate the high level overview - it's likely to get out of date with extension. we could move these comments to the code in the schema file and do a high level overview here (e.g. the separation of engine-owned tables, game-owned tables, and action-owned tables .

Reply:
good idea, done!